### ISSUE TITLE
Add provider_capability for NFS

### ISSUE THREAD

**Issue by Temikus** [mitchellh/vagrant#6781]
TL;DR: 
- Sometimes NFS isn't feasible nor needed (for example cloud providers).
- `SyncedFolders` always tries to set it up if host and remote machine support it.
- However, port forwarding is usually not supported, so the provisioning will fail.
- There are workarounds but can we have something proper like `provider_capability` implemented for NFS so providers can set whether nfs can be used with them?

In some cloud providers NFS is not supported nor needed and all syncing is done through Rsync.
However, the current Vagrant SyncedFolders implementation will try to set up NFS and while the local and remote system may be capable of it, the ports will not be forwarded correctly for it to work, so the provisioning will fail.

This can be worked around by setting an override entirely:

``` ruby
config.nfs.functional = false
```

, or setting the folder type explicitly:

``` ruby
congif.vm.synced_folder ".", "/vagrant",type: "rsync"
```

However, those are workarounds and need to be specifically set up, which makes people run into issues, for example (there are many more):
- vagrant-vcenter: frapposelli/vagrant-vcenter#5
- vagrant-rackspace: mitchellh/vagrant-rackspace#113
- vagrant-aws: https://github.com/mitchellh/vagrant-aws/issues/357
- vagrant-google: https://github.com/mitchellh/vagrant-google/issues/94

P.S. This idea was also proposed by @legal90 earlier in https://github.com/mitchellh/vagrant/issues/4011

**Comment by chrisroberts** [comment]
Hi there,

We would really like this, but this issue has been open for almost a year with no one working on it, Leaving it open is unfortunately making our issue count look higher than it is. I’m going to close this and if someone wants to work on it I’d still be open to a PR!

Cheers!

**Comment by Temikus** [comment]
For posterity sake, I'll note that adding:
```
@machine.config.nfs.functional = false
@machine.config.smb.functional = false
```
in `provider.rb`, does the equivalent of what I require.
